src/jit/codegenarmarch.cpp

```diff
@@ -1617,11 +1766,29 @@ void CodeGen::genCallInstruction(GenTreeCall* call)
         }
         else
         {
-            regNumber argReg = curArgTabEntry->regNum;
             genConsumeReg(argNode);
```

Comment:
The problem is that there are multiple registers, so `genConsumeReg()` needs to be called for each. That should take care of calling `genUnspillRegIfNeeded()`. The problem is that if you only call `genConsumeReg()` for the first register, then the liveness won't be updated for the other registers.